### ISSUE TITLE
Schema parsing error when using datastax Java driver 3.0.2

### ISSUE THREAD

**Issue by mjsaber** [scylladb/scylla#1538]
*Installation details*
Scylla version (or git commit hash): 1.1
Cluster size: standalone
OS (RHEL/CentOS/Ubuntu/AWS AMI): ubuntu 14.04

I have a following table:
> CREATE TABLE completed_repairs (
    k TEXT,
    v BIGINT,
    PRIMARY KEY (k)
) WITH compaction = { 'class' : 'LeveledCompactionStrategy' };

When I use following datastax cassandra driver connect to the cluster
```
<dependency>
    <groupId>com.datastax.cassandra</groupId>
    <artifactId>cassandra-driver-core</artifactId>
    <version>3.0.2</version>
</dependency>
```

I got following error:

> ERROR c.datastax.driver.core.TableMetadata - Error parsing schema options for table management.completed_repairs: Cluster.getMetadata().getKeyspace("management").getTable("completed_repairs").getOptions() will return null
> java.lang.IllegalArgumentException: Multiple entries with same key: class=LeveledCompactionStrategy and class=LeveledCompactionStrategy
> 	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:150) ~[jun-sandbox-playground.jar:na]
> 	at com.google.common.collect.RegularImmutableMap.checkNoConflictInBucket(RegularImmutableMap.java:104) ~[jun-sandbox-playground.jar:na]
> 	at com.google.common.collect.RegularImmutableMap.<init>(RegularImmutableMap.java:70) ~[jun-sandbox-playground.jar:na]
> 	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:254) ~[jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.TableOptionsMetadata.<init>(TableOptionsMetadata.java:138) ~[jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.TableMetadata.build(TableMetadata.java:137) ~[jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.SchemaParser.buildTables(SchemaParser.java:176) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.SchemaParser.buildKeyspaces(SchemaParser.java:128) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.SchemaParser.refresh(SchemaParser.java:64) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.ControlConnection.refreshSchema(ControlConnection.java:343) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.ControlConnection.tryConnect(ControlConnection.java:273) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:201) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:79) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1430) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.Cluster.init(Cluster.java:163) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:334) [jun-sandbox-playground.jar:na]
> 	at com.datastax.driver.core.Cluster.connect(Cluster.java:284) [jun-sandbox-playground.jar:na]
> 	at ConnectScylla.main(ConnectScylla.java:35) [jun-sandbox-playground.jar:na]

When I remove `WITH compaction = { 'class' : 'LeveledCompactionStrategy' };` then everything is fine, but the compaction strategy then becomes `SizeTieredCompactionStrategy`.

**UPDATE**
Tested datastax driver 2.1.10 works without ERROR, but in our case, we have tools that is shared between Cassandra 2, Cassandra 3 and Scylla, so we really want to use driver 3.0+ for all clusters.
Any suggestion?

Thanks,
Jun

**Comment by penberg** [comment]
The problem is that Scylla stores compaction strategy class in the compaction options field in system tables, which is Cassandra 3.0 behavior. This confuses the `TableOptionsMetadata` constructor check because it runs the Cassandra 2.x code path, which adds a second definition of compaction strategy class:

```java
        if (is300OrHigher)
            this.compaction = ImmutableMap.copyOf(row.getMap(COMPACTION, String.class, String.class));
        else {
            this.compaction = ImmutableMap.<String, String>builder()
                    .put("class", row.getString(COMPACTION_CLASS))
                    .putAll(SimpleJSONParser.parseStringMap(row.getString(COMPACTION_OPTIONS)))
                    .build();
        }
```

**Comment by penberg** [comment]
The difference is visible via ``cqlsh``.

Scylla:

```console
cqlsh:ks1> SELECT compaction_strategy_class, compaction_strategy_options FROM system.schema_columnfamilies WHERE keyspace_name = 'ks1';

 compaction_strategy_class | compaction_strategy_options
---------------------------+---------------------------------------
 LeveledCompactionStrategy | {"class":"LeveledCompactionStrategy"}
```

Cassandra 2.1.x:

```console
cqlsh:ks1> SELECT compaction_strategy_class, compaction_strategy_options FROM system.schema_columnfamilies WHERE keyspace_name = 'ks1';

 compaction_strategy_class                                       | compaction_strategy_options
-----------------------------------------------------------------+-----------------------------
    org.apache.cassandra.db.compaction.LeveledCompactionStrategy |                          {}
```

**Comment by penberg** [comment]
I sent a patch on the mailing list to fix this issue.

**Comment by penberg** [comment]
The fix will appear in the 1.3 release.
